lib/utils/class-alerts.php

```diff
@@ -0,0 +1,270 @@
+<?php
+
+namespace Automattic\VIP\Utils;
+
+class Alerts {
+	/**
+	 * The alerts service address
+	 *
+	 * @var string
+	 */
+	protected $service_address;
+
+	/**
+	 * The alerts service port
+	 *
+	 * @var string
+	 */
+	protected $service_port;
+
+	/**
+	 * Full path to the alerts service
+	 *
+	 * @var string
+	 */
+	protected $service_url;
+
+	/**
+	 * Instance of his Alerts class
+	 *
+	 * @var Alerts
+	 */
+	protected static $instance = null;
+
+	/**
+	 * Contructor
+	 *
+	 * Set to protected to prevent direct instantiation.
+	 * Use Alerts::get_instance() to get an instance of Alerts
+	 */
+	protected function __construct() {
+		if ( ! defined( 'ALERT_SERVICE_ADDRESS' ) || ! ALERT_SERVICE_ADDRESS ) {
+			throw new Exception( 'Missing alerts service host configuration in ALERT_SERVICE_ADDRESS constant' );
+		}
+
+		if ( ! defined( 'ALERT_SERVICE_PORT' ) || ! ALERT_SERVICE_PORT ) {
+			throw new Exception( 'Missing alerts service port configuration in ALERT_SERVICE_PORT constant' );
+		}
+
+		$this->service_address = ALERT_SERVICE_ADDRESS;
+		$this->service_port = ALERT_SERVICE_PORT;
+		$this->service_url = sprintf( 'http://%s:%s/v1.0/alerts', $this->service_address, $this->service_port );
+	}
+
+	/**
+	 * Send the alert
+	 *
+	 * @param array $body The alert message body
+	 *
+	 * @return array Response details from wp_remote_post
+	 */
+	protected function send( $body ) {
+		$response = wp_remote_post( $this->service_url, [
```

Comment:
Can we use the `vip_safe_wp_remote_request()` function instead?

Reply:
Changed in a08e0d406d6623be0ca464b6cd40289c7072dc96